web/src/lib/components/shared-components/combobox.svelte

```diff
@@ -53,15 +53,45 @@
   let selectedIndex: number | undefined;
   let optionRefs: HTMLElement[] = [];
   let input: HTMLInputElement;
+  let bounds: DOMRect | undefined;
+  let dropdownDirection: 'bottom' | 'top' = 'bottom';
+
   const inputId = `combobox-${id}`;
   const listboxId = `listbox-${id}`;
+  const dropdownOffset = 15;
+  const observer = new IntersectionObserver(
+    (entries) => {
+      const inputEntry = entries[0];
+      if (inputEntry.intersectionRatio < 1) {
+        isOpen = false;
+      }
+    },
+    { threshold: 0.5 },
+  );
 
   $: filteredOptions = options.filter((option) => option.label.toLowerCase().includes(searchQuery.toLowerCase()));
 
   $: {
     searchQuery = selectedOption ? selectedOption.label : '';
   }
 
+  $: position = calculatePosition(bounds);
+
+  onMount(() => {
+    observer.observe(input);
+    const scrollableAncestor = input.closest('.overflow-y-auto, .overflow-y-scroll');
+    scrollableAncestor?.addEventListener('scroll', onPositionChange);
+    window.visualViewport?.addEventListener('resize', onPositionChange);
+    window.visualViewport?.addEventListener('scroll', onPositionChange);
```

Comment:
Can't you apply the event listeners "svelte-style"? So on the respective component (or `svelte:window`) with `on:<event>` (e.g. `on:scroll`)?

Reply:
I don't think I can.

- `scrollableAncestor` - this contains the nearest ancestor component with a scrollbar and attaches a listener. This element is not inside of the combobox.
- `window.visualViewport` - the scroll and resize listeners specifically need to be attached to the `visualViewport`, to detect screen changes like a virtual keyboard appearing, or the browser app navigation appearing/disappearing as you scroll on a mobile phone. I don't think this can be done with `svelte:window`, but I'm open to ideas?

Reply:
Ah looks like you're right, sorry.